motionplan/rrt.go

```diff
@@ -66,16 +64,55 @@ func (plan *rrtPlanReturn) err() error {
 type rrtMaps struct {
 	startMap rrtMap
 	goalMap  rrtMap
+	optNode  *costNode // The highest quality IK solution
 }
 
-func initRRTMaps() *rrtMaps {
-	return &rrtMaps{
-		startMap: map[node]node{},
-		goalMap:  map[node]node{},
+func initRRTMaps(ctx context.Context, mp motionPlanner, goal spatialmath.Pose, seed []referenceframe.Input) *rrtPlanReturn {
+	rrt := &rrtPlanReturn{
+		rm: &rrtMaps{
+			startMap: map[node]node{},
+			goalMap:  map[node]node{},
+		},
 	}
+	seedNode := newCostNode(seed, 0)
+	rrt.rm.startMap[seedNode] = nil
+
+	// get many potential end goals from IK solver
+	solutions, err := mp.getSolutions(ctx, goal, seed)
+	if err != nil {
+		rrt.planerr = err
+		return rrt
+	}
+	mp.golog().Debugf("found %d IK solutions", len(solutions))
+
+	// the smallest interpolated distance between the start and end input represents a lower bound on cost
+	_, optimalCost := mp.opt().DistanceFunc(&ConstraintInput{StartInput: seed, EndInput: solutions[0].Q()})
+	rrt.rm.optNode = newCostNode(solutions[0].Q(), optimalCost)
+
+	// Check for direct interpolation for the subset of IK solutions within some multiple of optimal
+	// Since solutions are returned ordered, we check until one is out of bounds, then skip remaining checks
+	canInterp := true
+	// initialize maps and check whether direct interpolation is an option
+	for _, solution := range solutions {
+		if canInterp {
+			_, cost := mp.opt().DistanceFunc(&ConstraintInput{StartInput: seed, EndInput: solution.Q()})
+			if cost < optimalCost*defaultOptimalityMultiple {
+				if mp.checkPath(seed, solution.Q()) {
+					mp.golog().Debug("could interpolate directly to goal")
+					rrt.steps = []node{seedNode, solution}
+					return rrt
+				}
+			} else {
+				canInterp = false
+			}
+		}
+		rrt.rm.goalMap[newCostNode(solution.Q(), 0)] = nil
+	}
+	mp.golog().Debugf("failed to directly interpolate from %v to %v", seed, solutions[0].Q())
```

Comment:
This is just a thought I wanted to voice, not something actionable for this PR.  But we should eventually sit down and plan how we want to be logging things

Reply:
Agree. That'll probably be a company-wide overhaul at some point, honestly.

For now, I've been putting debug statements in the spots where I find myself putting `println` during dev debug.